datalad/support/sshconnector.py

```diff
@@ -110,59 +107,37 @@ def __init__(self, ctrl_path, sshri, identity_file=None,
-        # quoting rules, on other platforms args lists are passed directly and we need
-        # to take care of quoting ourselves
-        ctrlpath_arg = "ControlPath={}".format(ctrl_path if on_windows else sh_quote(str(ctrl_path)))
-        self._ssh_args = ["-o", ctrlpath_arg]
-        self.ctrl_path = Path(ctrl_path)
-        if self.sshri.port:
-            self._ssh_args += ['-p', '{}'.format(self.sshri.port)]
-
+        # arguments only used for opening a connection
+        self._ssh_open_args = []
+        # arguments for annex ssh invocation
+        self._ssh_args = []
+        self._ssh_open_args.extend(
+            ['-p', '{}'.format(self.sshri.port)] if self.sshri.port else [])
         if force_ip:
-            self._ssh_args.append("-{}".format(force_ip))
-        self._identity_file = identity_file
-        self._use_remote_annex_bundle = use_remote_annex_bundle
+            self._ssh_open_args.append("-{}".format(force_ip))
+        if identity_file:
+            self._ssh_open_args.extend(["-i", identity_file])
 
+        self._use_remote_annex_bundle = use_remote_annex_bundle
         # essential properties of the remote system
         self._remote_props = {}
-        self._opened_by_us = False
 
     def __call__(self, cmd, options=None, stdin=None, log_output=True):
-        """Executes a command on the remote.
+        raise NotImplementedError
 
-        It is the callers responsibility to properly quote commands
-        for remote execution (e.g. filename with spaces of other special
-        characters). Use the `sh_quote()` from the module for this purpose.
-
-        Parameters
-        ----------
-        cmd: str
-          command to run on the remote
-        options : list of str, optional
-          Additional options to pass to the `-o` flag of `ssh`. Note: Many
-          (probably most) of the available configuration options should not be
-          set here because they can critically change the properties of the
-          connection. This exists to allow options like SendEnv to be set.
+    def open(self):
+        raise NotImplementedError
 
-        Returns
-        -------
-        tuple of str
-          stdout, stderr of the command run.
-        """
+    def close(self):
+        raise NotImplementedError
 
-        # XXX: check for open socket once
-        #      and provide roll back if fails to run and was not explicitly
-        #      checked first
-        # MIH: this would mean that we would have to distinguish failure
-        #      of a payload command from failure of SSH itself. SSH however,
-        #      only distinguishes success and failure of the entire operation
-        #      Increase in fragility from introspection makes a potential
-        #      performance benefit a questionable improvement.
-        # make sure we have an open connection, will test if action is needed
-        # by itself
-        self.open()
+    @property
+    def _runner(self):
```

Comment:
unlikely anyone relies on it but especially since PR is against `maint` -- what is the reason for making `runner` protected ?

Reply:
It made no sense to expose it when building the base class. I you feel like this should be split, I am OK with that.